src/mint_json.cr

```diff
@@ -454,5 +454,19 @@ module Mint
         "node" => node(exception),
       }
     end
+
+    json_error MintJsonDependencyNotInstalled
+
+    def check_dependencies!(sources)
+      dependencies.each do |dependency|
+        next if dependency_exists?(sources, dependency.name)
+        raise MintJsonDependencyNotInstalled, {"name" => dependency.name}
+      end
+    end
+
+    def dependency_exists?(sources : Array(String), name : String)
+      pattern = /\.mint\/packages\/#{name}\/source/
```

Comment:
- the `source` directory might not be present for every package since it's configurable via `source-directories`
- I think at this point since we don't have a lock file It's enough to check if there is a directory for the package, then the sources are not even needed here
- once we have a lock file then we can do a more thorough check